### PR TITLE
Logging cloud for customers

### PR DESCRIPTION
**Pull request by pzet**
## Summary
Added logging to `CloudForCustomers` class methods.

## Importance
Messages help to trace the program execution and simplify debugging.

## Checklist

This PR:

- [ ] adds new tests (if appropriate)
- [ ] updates docstrings for any new functions or function arguments (if appropriate)
- [X] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: viadot/sources/cloud_for_customers.py [hunk 1]
```diff
@@ -205,6 +241,8 @@ def to_df(self, fields: List[str] = None, if_empty: str = "warn") -> pd.DataFram
         """
         records = self.to_records()
         df = pd.DataFrame(data=records)
+        if df.empty:
+            logger.warning("The newly created dataframe is empty.")
```
Comment (Trymzet) [inline viadot/sources/cloud_for_customers.py:245]: I would keep the convention we have of using camel case in this case (DataFrame).

**Review by Trymzet — CHANGES_REQUESTED** [review]
Minor changes

File: viadot/sources/cloud_for_customers.py [hunk 2]
```diff
@@ -75,7 +78,17 @@ def _to_records_report(self, url: str) -> List[Dict[str, Any]]:
...
         while url:
             response = self.get_response(url)
-            response_json = response.json()
+            try:
+                response_json = response.json()
+                logger.info("Data converted to JSON format.")
+            except requests.exceptions.JSONDecodeError:
+                logger.error(
+                    "Could not convert to JSON format. Content might be empty."
+                )
```
[the remark this answers is not in the document]
Comment (pzet) [reply viadot/sources/cloud_for_customers.py]: Changed to more generic messages: "Data fetched correctly." and "Data could not be fetched.", respectively.

File: viadot/sources/cloud_for_customers.py [hunk 3]
```diff
@@ -93,7 +106,15 @@ def _to_records_other(self, url: str) -> List[Dict[str, Any]]:
...
-            response_json = response.json()
+            try:
+                response_json = response.json()
+                logger.info("Data converted to JSON format.")
+            except requests.exceptions.JSONDecodeError as e:
+                logger.error("Could not convert to JSON format")
+
+            if not response_json:
+                logger.warning("It seems that JSON does not contain any data.")
+
```
[the remark this answers is not in the document]
Comment (pzet) [reply viadot/sources/cloud_for_customers.py]: Changed to more generic form: "It seems that the query produced no data."

File: viadot/sources/cloud_for_customers.py [hunk 4]
```diff
@@ -145,6 +167,7 @@ def response_to_entity_list(self, dirty_json: Dict[str, Any], url: str) -> List:
                         else:
                             new_entity[key] = object_of_interest
             entity_list.append(new_entity)
+        logger.info("JSON converted to list.")
```
[the remark this answers is not in the document]
Comment (pzet) [reply viadot/sources/cloud_for_customers.py]: Changed to "Data conversion successful."

File: viadot/sources/cloud_for_customers.py [hunk 5]
```diff
@@ -134,6 +155,7 @@ def response_to_entity_list(self, dirty_json: Dict[str, Any], url: str) -> List:
         metadata_url = self.change_to_meta_url(url)
         column_maper_dict = self.map_columns(metadata_url)
         entity_list = []
+        logger.info("Converting JSON to list...")
```
[the remark this answers is not in the document]
Comment (pzet) [reply viadot/sources/cloud_for_customers.py]: Log message is now removed.

File: viadot/sources/cloud_for_customers.py (hunk 1 above)
Comment (pzet) [reply viadot/sources/cloud_for_customers.py]: Corrected as suggested.

**Comment by Trymzet** [comment]
This needs a rebase as I see 14 changed files; should be about 2?

**Pull request closed without merging**